src/zenml/model/artifact_config.py

```diff
@@ -23,14 +23,14 @@
 from zenml.logger import get_logger
 
 if TYPE_CHECKING:
-    from zenml.model.model_config import ModelConfig
+    from zenml.model.model_version import ModelVersion
 
 
 logger = get_logger(__name__)
 
 
-class ArtifactConfig(BaseModel):
-    """Used to link a generic Artifact to the model version.
+class DataArtifactConfig(BaseModel):
+    """Used to link a data artifact to the model version.
 
     model_name: The name of the model to link artifact to.
     model_version: The identifier of the model version to link artifact to.
```

Comment:
```suggestion
    model_name: The name of the model to link data artifact to.
    model_version: The identifier of the model version to link data artifact to.
```

Maybe? Probably cascades too far to update all of these, however...